providers/saml.go

```diff
@@ -371,3 +363,40 @@ func PrintErrorStruct(err error) {
 		SAMLLogger.Debugf("%d: %s %s = %v\n", i, typeOfT.Field(i).Name, f.Type(), f.Interface())
 	}
 }
+
+func ReadNamesFromClaims(ForenameClaim, SurnameClaim string, claims map[string]interface{}) (forename, surname string) {
+	if ForenameClaim == "" {
+		ForenameClaim = DefaultForeNameClaim
+	}
+
+	if SurnameClaim == "" {
+		SurnameClaim = DefaultSurNameClaim
+	}
+
+	if val, ok := claims[ForenameClaim]; ok {
+		forename = val.(string)
+	}
+
+	if val, ok := claims[SurnameClaim]; ok {
+		surname = val.(string)
+	}
+
+	return
+}
+
+func ReadEmailFromClaims(emailClaim string, claims map[string]interface{}) (email string) {
+	// this is going to be a nightmare of slight differences between IDPs
+	// so lets make it configurable with a sensible backup
+	if emailClaim == "" {
+		emailClaim = DefaultEmailClaim
+	}
+
+	if _, ok := claims[emailClaim]; ok {
+		email = claims[emailClaim].(string)
+	} else if _, ok := claims["http://schemas.xmlsoap.org/ws/2005/05/identity/claims/"]; ok {
+		// fallback to WIF
+		email = claims[WIFUniqueName].(string)
```

Comment:
I think it would be good if we double check that claims map have `WIFUniqueName ` value before casting

Reply:
@tbuchaillot makes 100% sense

Reply:
@tbuchaillot done!